source/val/validate_constants.cpp

```diff
@@ -581,6 +581,59 @@ spv_result_t ValidateSpecConstantOp(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateConstantFunctionPointerINTEL(ValidationState_t& _,
+                                                  const Instruction* inst) {
+  // Check that the FunctionPointersINTEL capability is present
+  if (!_.HasCapability(spv::Capability::FunctionPointersINTEL)) {
+    return _.diag(SPV_ERROR_INVALID_CAPABILITY, inst)
+           << "OpConstantFunctionPointerINTEL requires FunctionPointersINTEL capability";
+  }
+
+  // Validate the result type is a function pointer
+  const auto result_type = _.FindDef(inst->type_id());
+  if (!result_type || result_type->opcode() != spv::Op::OpTypePointer) {
+    return _.diag(SPV_ERROR_INVALID_ID, inst)
+           << "OpConstantFunctionPointerINTEL Result Type <id> "
+           << _.getIdName(inst->type_id()) << " is not a pointer type";
+  }
+
+  // Check that the pointer points to a function type
+  const auto pointee_type = _.FindDef(result_type->GetOperandAs<uint32_t>(2));
+  if (!pointee_type || pointee_type->opcode() != spv::Op::OpTypeFunction) {
+    return _.diag(SPV_ERROR_INVALID_ID, inst)
+           << "OpConstantFunctionPointerINTEL Result Type <id> "
+           << _.getIdName(inst->type_id())
+           << " must be a pointer to function type";
+  }
+
+  // Validate that the function operand exists
+  const uint32_t function_id = inst->GetOperandAs<uint32_t>(2);
+  const auto function_inst = _.FindDef(function_id);
+  if (!function_inst) {
```

Comment:
Is this actually happening? This indicates there's no instruction for this id. At this point in the flow I don't think that's possible.